### PR TITLE
Implementation of UNC provider with support for forward and back slas…

### LINKED ISSUE #131: Source codes from UNC
Is there any support for UNC paths? I would like to host my sources on a network shared folder. but passing directly UNC path to the `-u` parameter results in a message saying that provider is not recognized.

### PR DESCRIPTION
**Pull request by maraf**
Fixes #131 .

### Checklist

- [x] I have included examples or tests
- [ ] I have updated the change log
- [ ] I am listed in the CONTRIBUTORS file
- [x] I have cleaned up the commit history (use rebase and squash)

### Changes proposed in this pull request:

- Added `UncProvider`.
- Added `IBackSlashSupport` and updated `Linker` to look if used implementation of `IProvider` implements this interface. If it does, it is asked whether back slashes are supported.

### CODE REVIEW

**Comment by maraf** [comment]
I'm not sure what I did wrong with appveyor. 
Any suggestions?

**Comment by GeertvanHorrik** [comment]
AppVeyor is new, it can be ignored. I will review this asap.

File: src/GitLink.Tests/Providers/UncProviderFacts.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
+﻿using GitLink.Providers;
```
Comment (GeertvanHorrik) [inline src/GitLink.Tests/Providers/UncProviderFacts.cs]: Please add the file header like the rest of the code

File: src/GitLink/Linker.cs [hunk 2]
```diff
@@ -252,5 +253,22 @@ private static bool LinkProject(Context context, Project project, string pdbStrF
...
             return true;
         }
+
+        private static string ReplaceSlashes(IProvider provider, string relativePathForUrl)
+        {
+            bool isBackSlashSupported = false;
+
+            // Check if provider is capable of determining whether to use back slashes or forward slashes.
+            IBackSlashSupport backSlashSupport = provider as IBackSlashSupport;
+            if (backSlashSupport != null)
```
Comment (GeertvanHorrik) [inline src/GitLink/Linker.cs]: Embed if statement logic inside { and }

File: src/GitLink/Providers/Interfaces/IBackSlashSupport.cs [hunk 3]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
```
Comment (GeertvanHorrik) [inline src/GitLink/Providers/Interfaces/IBackSlashSupport.cs]: Add file header

File: src/GitLink/Providers/UncProvider.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
+﻿using System;
```
Comment (GeertvanHorrik) [inline src/GitLink/Providers/UncProvider.cs]: Add file header

File: src/GitLink/Providers/UncProvider.cs [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+{
+    /// <summary>
+    /// A git provider for UNC network paths.
+    /// </summary>
+    public class UncProvider : ProviderBase, IBackSlashSupport
+    {
+        /// <summary>
+        /// A class with placeholder constants.
+        /// </summary>
+        public static class PlaceHolder
```
Comment (GeertvanHorrik) [inline src/GitLink/Providers/UncProvider.cs]: Not sure if embedding this inside a nested class is really necessary. Can you create 2 const members on the UncProvider instead?

File: src/GitLink/Providers/UncProvider.cs [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+        public override string RawGitUrl
+        {
+            get { return _rawUrl; }
+        }
+
+        public bool IsBackSlashSupported
+        {
+            get
+            {
+                if (RawGitUrl.StartsWith("//"))
```
Comment (GeertvanHorrik) [inline src/GitLink/Providers/UncProvider.cs]: Surround with { and }

**Comment by maraf** [comment]
I have made changes based on comments.
Please check.

**Comment by GeertvanHorrik** [comment]
Excellent. If you wish, please add yourself to CONTRIBUTORS as well. Then we can merge this PR.

**Comment by maraf** [comment]
I'm there.

**Comment by GeertvanHorrik** [comment]
Thanks a lot for your contribution, it's much appreciated!

**Comment by maraf** [comment]
Your welcome, GitLink is awesome.
Looking forward to next release.

**Pull request merged**